siac/walletcmd.go

```diff
@@ -457,10 +463,31 @@ func walletunlockcmd() {
 	if err != nil {
 		die("Reading password failed:", err)
 	}
-	qs := fmt.Sprintf("encryptionpassword=%s&dictonary=%s", password, "english")
-	err = post("/wallet/unlock", qs)
-	if err != nil {
-		die("Could not unlock wallet:", err)
+
+	errChan := make(chan error)
+	go func() {
+		qs := fmt.Sprintf("encryptionpassword=%s&dictonary=%s", password, "english")
+		errChan <- post("/wallet/unlock", qs)
+	}()
+
+	for {
+		select {
+		// when unlock completes, handle error
+		case err := <-errChan:
+			if err != nil {
+				die("Could not unlock wallet:", err)
+			}
+			fmt.Println("\nWallet unlocked")
+			return
+
+		// otherwise, display scan height every 3 seconds
+		case <-time.After(3 * time.Second):
+			var wg api.WalletGET
+			err := getAPI("/wallet", &wg)
+			if err != nil || !wg.Rescanning {
+				continue
+			}
+			fmt.Printf("\rScanned to height %v...", wg.ScanHeight)
```

Comment:
currently we print the height, but printing a percentage would be easy too. If we want to get fancy there's https://github.com/cheggaaa/pb